content/learn/book/your-first-game/_index.md

```diff
@@ -0,0 +1,685 @@
+        .add_system(scoreboard_system.system())
+```
+so it looks like this:
+```rs
+        .add_system(ball_movement_system.system())
+        .insert_resource(Scoreboard { score: 0 })
+        .add_system(scoreboard_system.system())
+        .run();
+```
+Next add a struct that represents our score right after the Ball struct:
+```rs
+struct Scoreboard {
+    score: usize,
+}
+```
+[usize](https://doc.rust-lang.org/std/primitive.usize.html) is a basic unsigned Rust type and represents how many bytes it takes to reference a location in memory.
+This type probably is overkill in our usecase, but it doesn't hurt either.
+
+
+Add the following lines just before the end to our setup function:
+
+```rs
+        commands.spawn(TextBundle {
+            text: Text {
+                sections: vec![
+                    TextSection {
+                        value: "Score: ".to_string(),
+                        style: TextStyle {
+                            font: asset_server.load("fonts/FiraSans-Bold.ttf"),
+                            font_size: 40.0,
+                            color: Color::rgb(0.5, 0.5, 1.0),
+                        },
+                    },
+                    TextSection {
+                        value: "".to_string(),
+                        style: TextStyle {
+                            font: asset_server.load("fonts/FiraMono-Medium.ttf"),
+                            font_size: 40.0,
+                            color: Color::rgb(1.0, 0.5, 0.5),
+                        },
+                    },
+                ],
+                ..Default::default()
+            },
+            style: Style {
+                position_type: PositionType::Absolute,
+                position: Rect {
+                    top: Val::Px(5.0),
+                    left: Val::Px(5.0),
+                    ..Default::default()
+                },
+                ..Default::default()
+            },
+            ..Default::default()
+        });
+```
+This looks different than what we had before, so let's go over it.  
+Here we are spawning a [TextBundle](https://docs.rs/bevy/latest/bevy/ui/entity/struct.TextBundle.html).
+
+A TextBundle contains many useful fields:
+
+```rs
+pub struct TextBundle {
+    pub node: Node,
+    pub style: Style,
+    pub draw: Draw,
+    pub visible: Visible,
+    pub text: Text,
+    pub calculated_size: CalculatedSize,
+    pub focus_policy: FocusPolicy,
+    pub transform: Transform,
+    pub global_transform: GlobalTransform,
+}
+```
+In our case we set the text field to be two `TextSections`.
+We do this so we can change the score number independently from the label in front of it.
+And all of the `..Default::default()` calls just make sure that the rest of the respective struct fields will get sensible values.
+
+Next we add a scoreboard system function:
+
+```rs
+fn scoreboard_system(scoreboard: Res<Scoreboard>, mut query: Query<&mut Text>) {
+    for mut text in query.iter_mut() {
```

Comment:
This is a problem in the example code, not your tutorial, but this should *really* use a marker component and a `.single_mut()` rather than iterating over all possible `Text` components.